### DISCUSSION TITLE
Enforcing Code Style

### DISCUSSION THREAD

**Discussion by stefannikolei** [SixLabors/ImageSharp#2650] [Ideas]
At the moment some style is configured as warnings, but not enforced.

For example this style is configured as warning but not used everywhere through the code base:

- csharp_style_var_... --> Usage of var is flagged as warning
- csharp_style_implicit_object_creation_when_type_is_apparent --> target type new
- csharp_style_prefer_range_operator 
- csharp_style_prefer_index_operator

We could fix all of this places in one PR. One solution to enforce this, would be to set the severity to error. What do you think about the severity? Keep it at warning or go to error?

**Comment by JimBobSquarePants** [comment]
As long as we are confident the analyzer doesn't produce false flags then I'm ok with making them errors.
